gossipd/seeker.c

```diff
@@ -963,18 +965,81 @@ static bool seek_any_unknown_nodes(struct seeker *seeker)
 	return true;
 }
 
+struct node_and_addrs {
+	struct node_id *id;
+	struct wireaddr *addrs;
+};
+
+/* Find a random node with an address in the announcement. */
+static struct node_and_addrs *get_random_node(const tal_t *ctx,
+				       struct seeker *seeker)
+{
+	struct gossmap *gossmap = gossmap_manage_get_gossmap(seeker->daemon->gm);
```

Comment:
This is better done by providing a "struct gossmap_node *gossmap_random_node(const struct gossmap *map);" in gossmap.c, which can use nodeidx_htable_pick.  Then iterate until NULL, or you find one with an address.